internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -94,15 +94,20 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if err != nil {
 		return ctrl.Result{}, r.updateErrorStatus(ctx, req.NamespacedName, cluster.Status, err)
 	}
+
+	deleted := map[types.UID]bool{}
 	for _, bundle := range cleanup {
-		for _, bundleDeployment := range bundleDeployments.Items {
-			if bundleDeployment.Labels[fleet.BundleLabel] == bundle.Name &&
-				bundleDeployment.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
-				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bundleDeployment)
-				err := r.Delete(ctx, &bundleDeployment) // nolint:gosec // does not store pointer
+		for _, bd := range bundleDeployments.Items {
+			if deleted[bd.GetUID()] {
+				continue
+			}
```

Comment:
Question: when is this happening? 
is it possible to have a `bundledeployment` that matches the condition to be deleted:
```go
bd.Labels[fleet.BundleLabel] == bundle.Name && bd.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace
```
more than once?

Reply:
Ah, I just wanted to skip the whole inner loop. But you're right, the existing `if` condition is probably not  significantly slower than `deleted[bd.GetUID()]`.